tar/storage/storage_write.c

```diff
@@ -306,8 +296,8 @@ storage_write_file(STORAGE_W * S, uint8_t * buf, size_t len,
 {
 	struct write_file_internal  * C;
 
-	/* If this is a dry run, return without doing anything. */
-	if (S->dryrun)
+	if (S == NULL)
+		/* Success for a dry run. */
```

Comment:
If you put a comment here, this becomes a multi-line if and you need braces.  But I would put the comment before the if.

Also, all we know here is that S is NULL.  It might be a dry run, or it might be for some completely other purpose.